### PR TITLE
SO-4382: Generic ConceptMap Request API test cases

### PR DESCRIPTION
**Pull request by molnarlaura**

### CODE REVIEW

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java [hunk 1]
```diff
@@ -31,32 +30,35 @@
...
 	}
 	
+	@Deprecated
 	public ConceptMapMappingSearchRequestBuilder filterByConceptMaps(Iterable<String> conceptMaps) {
 		return addOption(OptionKey.URI, conceptMaps);
 	}
 	
 	public ConceptMapMappingSearchRequestBuilder filterByConceptMapUri(ResourceURI conceptMapUri) {
-		return addOption(OptionKey.URI, conceptMapUri == null ? null : conceptMapUri.toString());
+		return filterByConceptMapUris(Set.of(conceptMapUri));
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: What happens if the parameter is `null`?

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java [hunk 2]
```diff
@@ -31,32 +30,35 @@
 		extends SearchPageableCollectionResourceRequestBuilder<ConceptMapMappingSearchRequestBuilder, ServiceProvider, ConceptMapMappings> 
 		implements SystemRequestBuilder<ConceptMapMappings> {
 	
+	//TODO: filterByConceptMap and filterByConceptMapUri should use a common uri because they use the same OptionKey
+	@Deprecated
 	public ConceptMapMappingSearchRequestBuilder filterByConceptMap(String conceptMap) {
 		return addOption(OptionKey.URI, conceptMap);
 	}
 	
+	@Deprecated
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: Why did you deprecate these filter methods? These will be used by the future REST API to access concept map content in a generic way.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java [hunk 3]
```diff
@@ -31,32 +30,35 @@
...
 	}
 	
 	public ConceptMapMappingSearchRequestBuilder filterByConceptMapUris(Iterable<ResourceURI> conceptMapUris) {
-		return addOption(OptionKey.URI, conceptMapUris == null ? null : FluentIterable.from(conceptMapUris).transform(ResourceURI::toString).toSet());
+		return addOption(OptionKey.URI, conceptMapUris == null ? null : conceptMapUris);
 	}
 	
 	public ConceptMapMappingSearchRequestBuilder filterByReferencedComponentId(String componentId) {
-		return addOption(OptionKey.REFERENCED_COMPONENT, componentId);
+		return filterByReferencedComponentIds(Set.of(componentId));
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: Same here, potential NPE.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java [hunk 4]
```diff
@@ -31,32 +30,35 @@
...
 	}
 	
 	public ConceptMapMappingSearchRequestBuilder filterByReferencedComponentIds(Iterable<String> componentIds) {
 		return addOption(OptionKey.REFERENCED_COMPONENT, componentIds);
 	}
 	
 	public ConceptMapMappingSearchRequestBuilder filterByMapTarget(String mapTarget) {
-		return addOption(OptionKey.MAP_TARGET, mapTarget);
+		return filterByMapTargets(Set.of(mapTarget));
```
Comment (cmark) [inline core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: And here.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java [hunk 5]
```diff
@@ -60,14 +60,20 @@
...
 			// TODO support proper refset SNOMED URIs as well
-			Set<ResourceURI> targetResources = search.getCollection(OptionKey.URI, String.class).stream()
-				.filter(ComponentURI::isValid)
-				.map(ComponentURI::of)
-				.map(ComponentURI::resourceUri)
-				.collect(Collectors.toSet());
 			
-			if (!targetResources.isEmpty()) {
-				return targetResources;
+			//Try to pars URIs to string, else return no results
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java]: Typo!

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java [hunk 6]
```diff
@@ -60,14 +60,20 @@
...
+			try {
+				Set<ResourceURI> targetResources = search.getCollection(OptionKey.URI, String.class).stream()
+						.filter(ComponentURI::isValid)
+						.map(ComponentURI::of)
+						.map(ComponentURI::resourceUri)
+						.collect(Collectors.toSet());
+				
+				if (!targetResources.isEmpty()) {
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java]: Just return the empty filter so it actually returns no results properly.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java [hunk 7]
```diff
@@ -60,14 +60,20 @@
...
+					return targetResources;
+				}
+			} catch (IllegalArgumentException e) {
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java]: How do you get an IllegalArgumentException here if the filter value collection is being filtered by `ComponentURI.isValid`? 😕

**Review by cmark — CHANGES_REQUESTED** [review]

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java (hunk 1 above)
Comment (molnarlaura) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: I removed the changes.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java (hunk 2 above)
Comment (molnarlaura) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: Deprecated the method with the ResourceURI.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java (hunk 3 above)
Comment (molnarlaura) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: I removed the changes.

File: core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java (hunk 4 above)
Comment (molnarlaura) [reply core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/request/ConceptMapMappingSearchRequestBuilder.java]: Same here.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java (hunk 7 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedConceptMapSearchRequestEvaluator.java]: Removed the try-catch block.

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**